docs/content/docs/deployment/memory/network_mem_tuning.md

```diff
@@ -0,0 +1,187 @@
+---
+title: "Network Buffer Tuning"
+weight: 100
+type: docs
+aliases:
+  - /deployment/memory/network_mem_tuning.html
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Network memory tuning guide
+
+## Overview
+
+Each record in flink is sent to the next subtask not individually but compounded in Network buffer,
+the smallest unit for communication between subtasks. Also, in order to keep consistent high throughput,
+Flink uses the network buffer queues (so called in-flight data) both on the output as well as on the input side. 
+In the result each subtask have an input queue waiting for the consumption and an output queue
+waiting for sending to the next subtask. Having a larger amount of the in-flight data means Flink can provide a
+higher throughput that's more resilient to small hiccups in the pipeline but it has negative effect for the
+checkpoint time. 
+
+The long checkpoint time issue can be caused by many things, one of those is checkpoint barriers
+propagation time. Checkpoint in Flink can finish only once all subtask receives all injected checkpoint
+barriers. In [aligned checkpoints]({{< ref "docs/concepts/stateful-stream-processing" >}}#checkpointing)
+those checkpoint barriers are traveling throughout the job graph along
+the network buffers and the larger amount of in-flight data the longer the checkpoint barrier propagation
+time. In [unaligned checkpoints]({{< ref "docs/concepts/stateful-stream-processing" >}}#unaligned-checkpointing) on the other hand, the more in-flight data, the larger the checkpoint size as
+all of the captured in-flight data has to be persisted as part of the checkpoint.
```

Comment:
```suggestion
Each record in Flink is sent to the next subtask compounded with other records in a *network buffer*,
the smallest unit for communication between subtasks. In order to maintain consistent high throughput,
Flink uses *network buffer queues* (also known as *in-flight data*) on the input and output side of the transmission process. 

Each subtask has an input queue waiting to consume data and an output queue
waiting to send data to the next subtask. Having a larger amount of in-flight data means that Flink can provide higher and more resilient throughput in the pipeline.  This will, however, cause longer checkpoint times. 

Checkpoints in Flink can only finish once all the subtasks receive all of the injected checkpoint
barriers. In [aligned checkpoints]({{< ref "docs/concepts/stateful-stream-processing" >}}#checkpointing), those checkpoint barriers are traveling throughout the job graph along with
the network buffers. The larger the amount of in-flight data, the longer the checkpoint barrier propagation time. In [unaligned checkpoints]({{< ref "docs/concepts/stateful-stream-processing" >}}#unaligned-checkpointing), the larger the amount of in-flight data, the larger the checkpoint size will be because all of the captured in-flight data has to be persisted as part of the checkpoint.
```